### PR TITLE
ARROW-12043: [Rust] [Parquet] Write FSB arrays

### PR DESCRIPTION
**Pull request by nevi-me**
Minor change to compute the levels for FSB arrays and write them out. Added a roundtrip test.

### CODE REVIEW

**Review by jorgecarleitao — APPROVED** [review]
LGTM. Great addition, thanks a lot @nevi-me !

File: rust/parquet/src/arrow/arrow_writer.rs [hunk 1]
```diff
@@ -156,6 +156,16 @@ fn write_leaves(
...
+        ArrowDataType::FixedSizeBinary(_) => {
+            let mut col_writer = get_col_writer(&mut row_group_writer)?;
+            write_leaf(
+                &mut col_writer,
+                array,
+                levels.pop().expect("Levels exhausted"),
+            )?;
+            row_group_writer.close_column(col_writer)?;
+            Ok(())
+        }
```
[the remark this answers is not in the document]
Comment (nevi-me) [reply rust/parquet/src/arrow/arrow_writer.rs:168]: lol, there isn't. Someone (won't name them) copied this chunk separately, thinking that it'll be different. Then they forgot to merge it when they realised that there was no difference.

I'll fix it shortly

File: rust/parquet/src/arrow/arrow_writer.rs [hunk 2]
```diff
@@ -189,11 +199,12 @@ fn write_leaves(
...
         )),
-        ArrowDataType::FixedSizeList(_, _)
-        | ArrowDataType::FixedSizeBinary(_)
-        | ArrowDataType::Union(_) => Err(ParquetError::NYI(
-            "Attempting to write an Arrow type that is not yet implemented".to_string(),
-        )),
+        ArrowDataType::FixedSizeList(_, _) | ArrowDataType::Union(_) => {
+            Err(ParquetError::NYI(
+                "Attempting to write an Arrow type that is not yet implemented"
+                    .to_string(),
```
Comment (alamb) [inline rust/parquet/src/arrow/arrow_writer.rs:205]:
```suggestion
                format!("Attempting to write an Arrow type {?} to parquet that is not yet implemented", array.data_type())
```

( I realize the error message is not changed by this PR I just saw it when looking)

**Review by alamb — APPROVED** [review]
Looks good to me -- thanks @nevi-me

File: rust/parquet/src/arrow/arrow_writer.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nevi-me) [reply rust/parquet/src/arrow/arrow_writer.rs]: I've fixed this

**Pull request closed without merging**